### DISCUSSION TITLE
Is there a way to use this library with Next.js ISR?

### DISCUSSION THREAD

**Discussion by dphuang2** [Pagefind/pagefind#496] [Q&A]
See [ISR documentation](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration) for explanation of ISR.

Main issue is that all the examples assume that entire website is known at build time but this isn't the case if you are using ISR.

Any examples/ideas?

**Comment by itsmatteomanf** [comment]
I guess you could try and create an index using the NodeJS APIs.

https://pagefind.app/docs/node-api/

This of course requires that all content is known at build time though, but that is gonna be a requirement for any at build, static, indexed search libraries.

**Comment by bglw** [comment]
Bringing this over to a discussion so it can live a bit longer 🙂

Echoing what @itsmatteomanf said — in essence if there's any stage where your site exists on disk with all content, that's the time to run Pagefind. If you're running fully dynamic through ISR or something in that vein, a service that crawls the website for search might be a better bet.

Integrating the Node API into a serverless or dynamic setup isn't something I have personally experimented with yet, so I'm definitely keen to hear from anyone who embarks down that path 🙂
